rita_common/src/payment_validator/mod.rs

```diff
@@ -92,6 +97,50 @@ impl Default for PaymentValidator {
     }
 }
 
+pub fn store_payment(pmt: PaymentTx, sent_from_us: bool) {
+    let mut lock = HISTORY.write().unwrap();
+    let neighbor;
+    let data = if sent_from_us {
+        neighbor = pmt.to;
+        &mut lock.successful_transactions_sent
+    } else {
+        neighbor = pmt.from;
+        &mut lock.successful_transactions_rec
+    };
+
+    if let std::collections::hash_map::Entry::Vacant(e) = data.entry(neighbor) {
```

Comment:
once again these can be distinct functions, it's net less lines and less chance of human error.

Reply:
fixed